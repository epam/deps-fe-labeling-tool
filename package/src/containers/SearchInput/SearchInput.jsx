import React from 'react'
import PropTypes from 'prop-types'
import { debounce } from '@/hocs/debounce'
import {
  Input, SearchIcon
} from './SearchInput.styles.js'

const RESOURCE_SEARCH_INPUT_PLACEHOLDER = 'Search'

const DebouncedSearchInput = debounce({
  getValue: (e) => e.target.value
})(Input)

const SearchInput = ({
  value = '',
  onChange
}) => {
  return (
    <DebouncedSearchInput
      placeholder={RESOURCE_SEARCH_INPUT_PLACEHOLDER}
      onChange={onChange}
      value={value}
      suffix={<SearchIcon />}
    />
  )
}

SearchInput.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
}

export {
  SearchInput
}
