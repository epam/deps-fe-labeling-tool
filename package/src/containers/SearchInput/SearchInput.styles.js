import { SearchOutlined } from '@ant-design/icons'
import styled from 'styled-components'
import { Input } from '@/components/Input'
import { COLORS } from '@/theme/theme.default'

const StyledInput = styled(Input)`
  position: relative;
  flex: 1 1 auto;
  line-height: 2.2rem;
  padding: 0.5rem 1.2rem;
  margin-right: 0.5rem;

  &::-webkit-input-placeholder {
    color: ${COLORS.SHADOW_1};
  }
`

const StyledSearchIcon = styled(SearchOutlined)`
  color: ${COLORS.GRAYSCALE_4};
`

export {
  StyledInput as Input,
  StyledSearchIcon as SearchIcon
}
