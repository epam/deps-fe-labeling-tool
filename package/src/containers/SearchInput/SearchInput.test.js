import React from 'react'
import { shallow } from 'enzyme'
import { SearchInput } from './SearchInput'

jest.mock('@/hocs/debounce', () => ({
  debounce: () => (Component) => Component
}))

describe('Components: SearchInput', () => {
  let defaultProps
  let wrapper

  beforeEach(() => {
    defaultProps = {
      value: 'search',
      onChange: jest.fn(() => 'test_value')
    }
    wrapper = shallow(<SearchInput {...defaultProps} />)
  })

  it('should render Input with correct props', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
