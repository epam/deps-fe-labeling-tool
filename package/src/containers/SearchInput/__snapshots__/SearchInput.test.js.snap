// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: SearchInput should render Input with correct props 1`] = `
<SearchInputstyles__StyledInput
  onChange={[MockFunction]}
  placeholder="Search"
  suffix={<SearchInputstyles__StyledSearchIcon />}
  value="search"
/>
`;
